pkg/kotsutil/kurl.go

```diff
@@ -59,3 +62,63 @@ func GetKurlRegistryCreds() (hostname string, username string, password string,
 
 	return
 }
+
+func SyncKurlKotsadmS3Secret(k8sConfigFlags *genericclioptions.ConfigFlags) error {
+	clientset, err := k8sutil.GetClientset(k8sConfigFlags)
+	if err != nil {
+		return errors.Wrap(err, "failed to get clientset")
+	}
+
+	s3Secret, err := clientset.CoreV1().Secrets("default").Get(context.TODO(), "kotsadm-s3", metav1.GetOptions{})
+	if err != nil {
+		return errors.Wrap(err, "failed to get s3 secret")
+	}
+
+	s3Endpoint := string(s3Secret.Data["endpoint"])
+
+	if strings.Contains(s3Endpoint, "rook-ceph") {
+		rookSecret, err := clientset.CoreV1().Secrets("rook-ceph").Get(context.TODO(), "rook-ceph-object-user-rook-ceph-store-kurl", metav1.GetOptions{})
+		if err != nil {
+			return errors.Wrap(err, "failed to get rook-ceph-object-user-rook-ceph-store-kurl secret")
+		}
+		s3Secret.Data["access-key-id"] = rookSecret.Data["AccessKey"]
+		s3Secret.Data["secret-access-key"] = rookSecret.Data["SecretKey"]
+
+		rookService, err := clientset.CoreV1().Services("rook-ceph").Get(context.TODO(), "rook-ceph-rgw-rook-ceph-store", metav1.GetOptions{})
+		if err != nil {
+			return errors.Wrap(err, "failed to get rook-ceph-rgw-rook-ceph-store service")
+		}
+		s3Secret.Data["object-store-cluster-ip"] = []byte(rookService.Spec.ClusterIP)
+
+		_, err = clientset.CoreV1().Secrets("default").Update(context.TODO(), s3Secret, metav1.UpdateOptions{})
+		if err != nil {
+			return errors.Wrap(err, "failed to update s3 secret from rook secret")
+		}
+
+		return nil
+	}
+
+	if strings.Contains(s3Endpoint, "minio") {
+		minioSecret, err := clientset.CoreV1().Secrets("minio").Get(context.TODO(), "minio-credentials", metav1.GetOptions{})
+		if err != nil {
+			return errors.Wrap(err, "failed to get minio-credentials secret")
+		}
+		s3Secret.Data["access-key-id"] = minioSecret.Data["MINIO_ACCESS_KEY"]
+		s3Secret.Data["secret-access-key"] = minioSecret.Data["MINIO_SECRET_KEY"]
+
+		minioService, err := clientset.CoreV1().Services("minio").Get(context.TODO(), "minio", metav1.GetOptions{})
+		if err != nil {
+			return errors.Wrap(err, "failed to get minio service")
+		}
+		s3Secret.Data["object-store-cluster-ip"] = []byte(minioService.Spec.ClusterIP)
+
+		_, err = clientset.CoreV1().Secrets("default").Update(context.TODO(), s3Secret, metav1.UpdateOptions{})
+		if err != nil {
+			return errors.Wrap(err, "failed to update s3 secret from minio secret")
+		}
+
+		return nil
+	}
+
+	return errors.New("unsupported object store")
```

Comment:
does endpoint have any secrets in it?  could we add it to the error?

Reply:
it does not, good idea